repos/system_upgrade/el7toel8/models/initrd.py

```diff
@@ -0,0 +1,134 @@
+from leapp.models import Model, fields
+from leapp.topics import BootPrepTopic, SystemInfoTopic
+from leapp.utils.deprecation import deprecated
+
+
+class DracutModule(Model):
+    """
+    Specify a dracut module that should be included into the init ramdisk
+
+    The specified dracut module has to be compatible with the target system.
+
+    See the description of UpgradeInitrdTasks and TargetInitrdTasks for more
+    information about the role of init ramdisks in the in-place upgrade
+    process.
+    """
+    topic = BootPrepTopic
+
+    name = fields.String()
+    """
+    Name of the dracut module that should be added (--add option of dracut)
+    """
+
+    # FIXME: proposed: remove the module_path!!!! TargetUserSpaceInitrdEnvTasks
+    # problem(?): the modules could be used for the target init ramdisk as
+    # well. That means, if the module is not installed by an rpm during the
+    # upgrade, user has to create separate actor that copies the files under
+    # expected path - and such an actor has to be executed before the target
+    # init ramdisk is influenced. The last condition is relatively ok as the
+    # target init ramdisk is modified during the FinalizationPhase, but
+    # thinking about that, it does't seem to be so comfortable change when
+    # people have to create another additional actor for that. I will leave
+    # this for now untouched, so others could think about that as well to decide
+    # the change. (@oamg/developers, @vinzenz)
+    module_path = fields.Nullable(fields.String(default=None))
+    """
+    module_path specifies dracut modules that are supposed to be copied
+
+    If the path is not set, the given name will just be activated. IOW,
+    if the dracut module is stored outside the /usr/lib/dracut/modules.d/
+    directory, set the absolute path to it, so leapp will manage it during
+    the upgrade to ensure the module will be added into the init ramdisk.
+
+    The module has to be stored on the local storage. In such a case, it is
+    recommended to store it into the 'files' directory of an actor generating
+    this object.
+    """
```

Comment:
@vinzenz (@oamg/administrators): Thinking about the DracutModule.module_path removal, I am not sure whether it's good path. I see pros & cons.

Pros for removal: more clean design around models (similar functionality we have for containers already, so user could resolve it for upgrade init ramdisk just by generating one additional message).

Cons: for target init ramdisk, behaviour will be different. not sure what is really typical on production systems, but in case a special dracut modules are handled out of rpms, user will have to cover filecopy by additional extra actor.

I will leave it for discussion. I can work on additional parts of this PR in the meanwhile.

Reply:
We are going to keep the field in the model.